ledger-service/cli-opts/src/main/scala/cliopts/Metrics.scala

```diff
@@ -6,9 +6,28 @@ package com.daml.cliopts
 import com.daml.metrics.MetricsReporter
 import scopt.OptionDef
 
-import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.duration.{Duration, FiniteDuration, NANOSECONDS}
+import scala.util.Try
 
 object Metrics {
+  private case class DurationFormat(unwrap: FiniteDuration)
+
+  private implicit val scoptDurationFormat: scopt.Read[DurationFormat] = scopt.Read.reads {
+    duration =>
+      Try {
+        Duration(duration)
+      }.orElse(Try {
+        Duration.fromNanos(
+          java.time.Duration.parse(duration).toNanos
+        )
+      }).flatMap(duration =>
+        Try {
+          if (!duration.isFinite)
+            throw new RuntimeException(s"Input duration $duration is not finite")
```

Comment:
```suggestion
            throw new IllegalArgumentException(s"Input duration $duration is not finite")
```
seems to match what scopt throws itself